source/security/authentication.txt

```diff
@@ -273,6 +273,23 @@ URI to ``"MONGODB-AWS"``, as shown in the following example:
    to retrieve credentials only from an EC2 instance, ensure that the
    environment variables are not set.
 
+Error Handling
+--------------
+
+You can handle authentication errors by catching ``mongocxx::exception``
+errors. Inspect the associated error code by using the ``code()`` method, as
+shown in the following example:
```

Comment:
```suggestion
errors, as shown in the following example:
```
Recommend that we avoid encouraging users to inspect `.code()`, unless they _really+ need to know the specific error code/category and programmatically handle them individually, until we've addressed CXX-834. As demonstrated by the [Invalid Auto Encryption Options](https://mongocxx.org/api/mongocxx-4.0.0/topic-mongocxx-examples-clients.html#invalid-auto-encryption-options) example, error codes are currently overloaded with no reliable method to distinguish the different domains (server? C++ Driver? libmongoc? libmongocrypt?). The most user-friendly method atm is to inspect the error message string with `.what()` instead.